src/Core/Manifest/PrioritySorter.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+namespace SilverStripe\Core\Manifest;
+
+use SilverStripe\Core\Injector\Injectable;
+
+/**
+ * Class PrioritySorter
+ * @package SilverStripe\Core\Manifest
+ */
+class PrioritySorter
+{
+    use Injectable;
+
+    /**
+     * The key that is used to denote all remaining items that have not
+     * been specified in priorities
+     * @var string
+     */
+    protected $restKey = '...rest';
+
+    /**
+     * A map of variables to their values
+     * @var array
+     */
+    protected $variables = [];
+
+    /**
+     * @var array
+     */
+    protected $items;
+
+    /**
+     * @var array
+     */
+    protected $priorities;
+
+    /**
+     * @var array
+     */
+    protected $names;
+
+    /**
+     * If not specified in priorities, put this item at the top (can be a variable)
+     *
+     * @var string
+     */
+    protected $defaultTop;
+
+    /**
+     * PrioritySorter constructor.
+     * @param array $items
+     * @param array $priorities
+     */
+    public function __construct(array $items = [], array $priorities = [])
+    {
+        $this->setItems($items);
+        $this->priorities = $priorities;
+    }
+
+    /**
+     * Sorts the items and returns a new version of $this->items
+     *
+     * @return array
+     */
+    public function getSortedList()
+    {
+        $this->addVariables();
+
+        // Find all items that don't have their order specified by the config system
+        $unspecified = array_diff($this->names, $this->priorities);
+
+        if ($this->restKey && !empty($unspecified)) {
+            $this->includeRest($unspecified);
+        }
+
+        if ($this->defaultTop) {
```

Comment:
Suggested earlier, I don't think we need a defaultTop if variable is included in priority in correct place.